### PR TITLE
Mainline boats mod

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Content "Mainline Boats mod"```

#### Purpose of change
Now that boats push fish out the way, and lakes are in, it makes sense to mainline boats, as there are no glaring problems with them ( and bridges will get worked on eventually )

#### Describe the solution
Mainlined the mod

#### Describe alternatives you've considered
N/A

#### Additional context
N/A

### CODE REVIEW

File: data/json/items/tools.json [hunk 1]
```diff
@@ -74,6 +74,21 @@
       "type": "transform"
     }
   },
+  {
+    "id": "hand_pump",
+    "type": "TOOL",
+    "symbol": "/",
+    "color": "blue",
+    "name": "hand pump",
+    "name_plural": "hand pumps",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: Redundant.

File: data/json/items/tools.json [hunk 2]
```diff
@@ -74,6 +74,21 @@
...
+    "description": "This pump is suitable for pumping air into inflatable objects.",
+    "price": 400,
+    "material": [ "aluminum", "plastic" ],
+    "weight": 113,
+    "volume": 1,
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: I think that's too small. At least 0.5 liters, perhaps?

File: data/json/items/tools.json [hunk 3]
```diff
@@ -74,6 +74,21 @@
...
+    "cutting": 6,
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: Hand pump doesn't have any edge to provide cutting damage, I believe.

File: data/json/overmap_terrain.json [hunk 4]
```diff
@@ -47,6 +47,27 @@
     "sym": "O",
     "color": "white"
   },
+  {
+    "//": "Add the following to overmap_terrain.json in /data/json/",
```
Comment (Night-Pryanik) [inline data/json/overmap_terrain.json]: Now it became useless.

File: data/json/items/tools.json (hunk 1 above)
Comment (davidpwbrown) [reply data/json/items/tools.json]: The plural name is redundant? it seems to fit the same place as the other items and their plural names

Comment (anothersimulacrum) [reply data/json/items/tools.json]: The `s` is automatically appended if a plural name is missing, so there is no point having a plural name specify that.

**Comment by ZhilkinSerg** [comment]
We need to update some tests:

```
Boat mod is loaded correctly or not at all
-------------------------------------------------------------------------------
mod_test.cpp:10
...............................................................................

mod_test.cpp:16: FAILED:
  REQUIRE( !item::type_is_defined( "inflatable_boat" ) )
with expansion:
  false

===============================================================================
test cases:     153 |     151 passed | 2 failed
assertions: 2078839 | 2078837 passed | 2 failed

```

**Comment by davidpwbrown** [comment]
Right so I think this is the way? Ive deleted mod_test.cpp as thats only purpose was to check if boat mod was active, which is no longer needed, and added all the boats to the vehicle_drag test .

**Pull request merged**